macros/Compat.ejs

```diff
@@ -1,66 +1,399 @@
-Web/CSS/@font-face                                      css/at-rules/font-face.json                                           {{Compat("css/at-rules/font-face")}}
-
-Parameters:
-
-$0 – json file in the repository (without ".json")
-$1 – (optional) Node in that json 
+Retrieves MDN compat data for a given feature
+
+Differentiates between aggregate compat tables and feature compat tables.
+Aggregate tables use basic_support of the underlying features and have no notes
+Feature tables display sub features and display notes
+
+Parameters
+
+  $0 – A query string indicating for which feature to retrieve compat data for.
+
+Example calls
+
+  {{compat("webextensions.api.alarms")}}
+  {{compat("webextensions.api.alarms.Alarm")}}
+  {{compat("webextensions.manifest")}}
+  {{compat("webextensions.manifest.applications")}}
+  {{compat("css.properties.background-attachment")}}
+  {{compat("css.selectors.any-link")}}
+  {{compat("css.at-rules.font-face")}}
+  {{compat("http.headers.Cache-Control")}}
+  {{compat("http.status.404")}}
+  {{compat("http.method.CONNECT")}}
+  {{compat("api.WebGL2RenderingContext")}}
+  {{compat("api.WebGL2RenderingContext.beginQuery")}}
+  {{compat("html.elements.input")}}
+
+  {{compat("css.properties.background-*")}}
+  (not implemented, just an idea for more advanced aggregate queries)
 
 */
 
-var githubURL = "https://raw.githubusercontent.com/mdn/browser-compat-data/master/";
-var locale = env.locale;
-var output = "";
-var file = $0 + ".json";
-var leaf = $1;
-var url = githubURL + file;
+var bcd = require('mdn-browser-compat-data');
 
-var noDataFoundl10n = mdn.localString({
-    "en-US": 'No compatibility data found. Please contribute data for "' + env.title + '" to the <a href="https://github.com/mdn/browser-compat-data">MDN compatibility data repository</a>.',
-    "fr"   : "Aucune donnée de compatibilité trouvée"
+var query = $0;
+var output = '';
+
+var compatStrings = string.deserialize(template("L10n:CompatTable"));
+var localize = mdn.getLocalString;
+
+var s_no_data_found = mdn.localString({
+    'en-US': `No compatibility data found. Please contribute data for "${query}" to the <a href="https://github.com/mdn/browser-compat-data">MDN compatibility data repository</a>.`
 });
 
+const desktopBrowsers = {
+  chrome: "Chrome",
+  firefox: "Firefox",
+  edge: "Edge",
+  ie: "Internet Explorer",
+  opera: "Opera",
+  safari: "Safari",
+  servo: "Servo"
+};
+
+const mobileBrowsers = {
+  webview_android: "Android",
+  chrome_android: "Chrome for Android",
+  edge_mobile: "Edge mobile",
+  firefox_android: "Firefox for Android",
+  ie_mobile: "IE mobile",
+  opera_android: "Opera Android",
+  safari_ios: "iOS Safari",
+}
+
+var notesArray = [];
 
-if ($0) {
-  try {
-    var data = mdn.fetchJSONResource(url);
-  } catch (e) {
-    output = 'File "' + file + '" not found or malformed.'
+/*
+Write the table header.
+*/
+function writeTableHead(environment, browserNames) {
```

Comment:
I updated the function comments throughout the file.

I renamed `environment` to `browserPlatformType`.